packages/abi-coder/src/interface.ts

```diff
@@ -173,4 +177,8 @@ export default class Interface {
 
     return this.abiCoder.encode(fragment.outputs, values);
   }
+
+  updateLoggedTypes(id: string, loggedTypes: JsonAbiLogFragment[]) {
+    this.logTypeHash[id] = loggedTypes;
+  }
```

Comment:
This method name suggests that it will update the `loggedTypes` variable, which can be confusing (_when looked at from the outside_) since it modifies the `logTypeHash` instead.

Following my first suggestion of renaming the `logTypeHash` variable to `externalLoggedTypes`, this could follow the same rationale and be something like this: 

```ts
  /*
    Called from `<base-invocation-scope.ts>.addContracts()` method,
    used for saving the `loggedTypes` ABI definitions referring to
    external contracts being added, needed for proper log decoding.
  */
  addExternalLoggedTypes(contractId: string, loggedTypes: JsonAbiLogFragment[]) {
    this.externalLoggedTypes[contractId] = loggedTypes;
  }
```